Сетка страниц и крупных блоков на флексах

TromiUss opened the pull request:
---
:mortar_board: [Базовая стилизация всех страниц](https://up.htmlacademy.ru/htmlcss/36/user/2300497/tasks/8)

:boom: https://htmlacademy-htmlcss.github.io/2300497-sedona-36/3/

---
:mortar_board: [Сетка страниц и крупных блоков на флексах](https://up.htmlacademy.ru/htmlcss/36/user/2300497/tasks/10)

TromiUss committed:
мелкие недочеты

Исправил оставшиеся недочеты

TromiUss committed:
Css по макету

Сделал, как по макету ( тренировка )

TromiUss committed:
Styles

TromiUss committed:
main

Добавил правки текста и кнопок в main

TromiUss committed:
footer

Текст, и кнопки футера

TromiUss committed:
footer-back

другая фотография

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,44 @@
+  font-display: swap;
+}
+
+@font-face {
```
Дубль правила. Нужно удалить. Если мы отдельно задаем italic, то только тогда, когда у нас для этого есть отдельный файл шрифта с курсивным начертанием. Вы пытаетесь для этого включить тот же шрифт. Не надо, в таком случае это браузер и так сделает.

efiand replied:
https://gwfh.mranftl.com/fonts - вот сервис для генерации woff2 из гугловских шрифтов по заданным настройкам. Если нужен курсив, создайте его здесь. Но я не припомню, где в Седоне курсив.

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,44 @@
+  font-family:"PT Sans" ;
+  font-style: normal;
+  font-weight: 400;
+  src: url(../fonts/ptsans-400.woff2) format(woff2);
```
Критерий о единообразии учит нас еще и тому, что все строковые данные стоит писать одинаково, то есть в кавычках.

![изображение](https://user-images.githubusercontent.com/26799424/216941362-dc6893bc-de08-4a84-b246-e71e9b5197b5.png)

Вам сейчас одинаково сложно работать с обоими примерами на скриншоте, а мне уже гораздо проще понимать правильно подсвеченные код из первого варианта, где цвет ЗНАЧЕНИЙ (коричневый) не подсвечен цветом СВОЙСТВ (голубым). И многим другим разработчикам, с кем в команде можете потом столкнуться, тоже нравится, когда все сделано по оговоренным правилам.

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,44 @@
+body {
+  font-family: "PT Sans", sans-serif;
+  font-size: 16px;
+  line-height: 24px;
```
Лучше указывать базовый шрифт множителем (`1.5`), так как в случае добавления каких-либо виджетов, где текст рассчитан на настройки проекта, очень крупный многострочный текст будет слипаться. Если задать множитель, то для любого текста размером, например, 48px, высота строки будет 72px, а не 24px, и ничего не слипнется. Для текста размером 16px это будут те же `24px`, ничего не потеряем.

efiand commented on images/YT.svg:
```diff
@@ -0,0 +1,3 @@
+<svg width="23" height="18" viewBox="0 0 23 18" fill="none" xmlns="http://www.w3.org/2000/svg">
```
По критерию имена файлов с маленькой буквы, поэтому, чтобы гит смог это запушить без танцев с бубном, придумайте новое имя. Например, `youtube`

efiand commented on index.html:
```diff
@@ -9,45 +9,46 @@
           </li>
           <li class="user-item">
-            <a class="user-link" href="#!">Хочу сюда</a>
+            <a class="user-link user-link-button" href="#!">Хочу сюда</a>
```
Эта коричневая кнопка по базовой стилизации ничем не отличается от кнопок в каталоге, например. Можно подумать о подходе, когда все типовые текстовые кнопки сделаны классом `button` и дополнительным `button-чтототам` для стилизации отличия некоторых кнопок

efiand commented on index.html:
```diff
@@ -100,7 +101,8 @@ <h3 class="services-title">Сувениры</h3>
         <b class="hotels-question">Заинтересовались?</b>
-        <p class="hotels-proposal">Укажите предполагаемые даты поездки, и мы покажем вам лучшие предложения гостиниц в Седоне</p>
+        <p class="hotels-proposal">Укажите предполагаемые даты поездки, и мы покажем вам лучшие предложения гостиниц в
+          Седоне</p>
```
Чтобы такого не происходило, можно переформатировать вручную
```html
       <p class="hotels-proposal">
         Укажите предполагаемые даты поездки, и мы покажем вам лучшие предложения гостиниц в Седоне
       </p>
```
тогда не будет при повторном автоформатировании бить по середине фразы

efiand commented on index.html:
```diff
@@ -116,21 +118,23 @@ <h2 class="content-title">Подпишитесь на рассылку</h2>
+        </ul>
+        <a class="tel" href="tel:+78128121212">+7 (812) 812-12-12</a>
+        <a href="https://htmlacademy.ru" class="html-academy-link">
+          <img src="images/htmlacademy-logo.svg" width="114.9" height="32.9" alt="Логотип HTML Academy">
```
![изображение](https://user-images.githubusercontent.com/26799424/216945342-7b674b54-2a2a-47eb-8c80-9eefcba9da8d.png)

Вполне адекватные размеры в самой картинке. На баг макета не ведитесь

efiand commented on index.html:
```diff
@@ -116,21 +118,23 @@ <h2 class="content-title">Подпишитесь на рассылку</h2>
+      <section class="footer-links">
+        <ul class="social-network">
+          <li class="social-network-item">
+            <a class="social-network-link wk" href="https://vk.com/htmlacademy"><span class="visually-hidden">Вк.</span></a>
```
`v` вместо `w` было бы логичнее

efiand commented on index.html:
```diff
@@ -116,21 +118,23 @@ <h2 class="content-title">Подпишитесь на рассылку</h2>
+            <a class="social-network-link telegram" href="https://t.me/htmlacademy "><span class="visually-hidden">Телеграм.</span></a>
+          </li>
+          <li class="social-network-item">
+            <a class="social-network-link YouTube" href="https://www.youtube.com/user/htmlacademyru"><span class="visually-hidden">YouTube.</span></a>
```
Классы именуем единообразно - строчными буквами

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+  line-height: 24px;
+  color: #242424;
+  background-color: #ffffff;
+
```
Что-то не так, наверное, с автоформаттером. Какой используете, стандартный или Prettier, или еще какой-то? Не должно быть таких "дырок". Если и должно, то везде, но такого ни в одном кодгайде я не видел

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
 }
+*/
+@font-face {
+  font-family: "PT sans";
```
Принято называть этот шрифт `PT Sans`, с большой буквы

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+}
+
+.header-link{
+  font-family: "PT sans";
```
Во-первых, на ссылках шрифт наследуется. Во-вторых, если вы задаете шрифт, то почему не указано семейство в конце?

Если вы используете класс на `<button>`, `<select>` или `<input>`, можно написать `font-family: inherit` - тогда в явном виде будет наследоваться та колбаса вариантов, что один раз уже прописали в `body {}`. Для большинства других тегов наследование и так происходит. Наследуется или нет, в devtools видно

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+
+.header-link{
+  font-family: "PT sans";
+  font-style: normal;
```
Он и так по всему проекту normal, не надо это писать везде. `normal` пишем только тогда, когда весь шрифт в блоке курсивный, а один элемент - нет.
Либо для универсальных классов, которые используются в любых блоках. Это чисто шапочные стили, тут нет курсива, так что свойство мусорное.

TromiUss replied:
При удалении этой строки в user-count строчки в списке header наехали друг на друга

efiand replied:
![изображение](https://user-images.githubusercontent.com/26799424/217084336-590b224a-b854-4b25-8690-ac66ad38e8b2.png)
![изображение](https://user-images.githubusercontent.com/26799424/217084942-d8899191-b7e6-4d27-9163-d53a35e58e19.png)

Не понял, о чем речь, свойство `font-style: normal;` не влияет вообще ни на что.

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+  font-style: normal;
+  font-weight: 700;
+  font-size: 20px;
+  line-height: 24px;
```
Если вы не перепишете в `body` на множитель, то в этом свойстве также нет смысла, ибо оно наследуется. Сейчас браузер почем зря переопределяет на то же самое:

![изображение](https://user-images.githubusercontent.com/26799424/216947912-81ebc9b2-2243-40ee-b384-053887520211.png)

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+  font-weight: 700;
+  font-size: 20px;
+  line-height: 24px;
+  color: inherit;
```
👍🏻 А вот это правильный подход для ссылок, они ведь черные не сами по себе, а просто в стиль родительского текста

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+  font-size: 20px;
+  line-height: 24px;
+  color: inherit;
+  text-align: center;
```
Бесполезное свойство здесь, вы поймете это на сетках

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+  background-position: center;
+  background-repeat: no-repeat;
+  text-decoration: none;
+  color: white;
```
Цвет принято задавать единообразно по проекту. Например, в hex-формате `#rrggbb`.
К тому же этот цвет в большей степени относится к спану, ведь именно он на зеленом фоне. Чтобы никогда не произошло такого, что на темном фоне вдруг оказался темный цвет, а на светлом - светлый, `всегда задавайте цвет текста в том же правиле, что и цвет фона`

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+
+.user-link-search {
+  background-image: url(../images/search.svg);
+  background-position: center;
```
Позицию и повторяемость фона лучше задавать на общем классе `user-link`, чтобы не дублировать свойства. По двум иконкам трудно понять преимущества, но представьте, что их 7-8, такие блоки бывают. Для того и нужны общие классы, чтобы под ними описывать общие свойства. А под дополнительными - только отличия

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+.hero {
+
+  background: url(../images/index-background.jpg) no-repeat center;
+  margin: 0%;
```
А почему `0` именно процентов, а не пикселей? Или `em`?
В общем, у нуля не принято ставить единицы измерения. Просто `0`. И в кодгайде академии это прописано.

Кроме того, у секций и так нет маргинов. Если привыкать мусорить, можно зайти очень далеко.
Почему, например, не указать бы снова шрифт, бордер нулевой и т. п.? Почему именно эти мусорные свойства, а не какие-то другие? Вывод: не пишите вообще того, что не решает никаких задач

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+  background: url(../images/index-background.jpg) no-repeat center;
+  margin: 0%;
+  padding: 0%;
+  background-color: #242424;
```
![изображение](https://user-images.githubusercontent.com/26799424/216949997-c3dac535-9ad4-494b-b4a2-7b34bad7aeda.png)

👍🏻 Надо сказать, это хорошая идея для замешающего цвета. Я думал, а какой же из проектных цветов подобрать на замену: голубой, коричневый и белый сделают часть эмблемы невидимой.

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+  background-color: #242424;
+  background-size: 100% auto;
+  text-align: center;
+  background-attachment: fixed;
```
Это свойство проявилось бы только в том случае, если бы мы задали отдельно этому блоку свою прокрутку и напихали текста. Но так как все равно при добавлении текста блок просто расширится, в этом нет никакого смысла.

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+  margin: 0%;
+  padding: 0%;
+  background-color: #242424;
+  background-size: 100% auto;
```
![изображение](https://user-images.githubusercontent.com/26799424/216951102-839a532e-7fb1-41ab-8d37-4cde192657a3.png)
Очень часто те или иные блоки в течение жизни проекта получают новый контент, и фона при таком размере может не хватать. Лучше всего задавать значение `cover` - при небольшом переполнении спасёт:

![изображение](https://user-images.githubusercontent.com/26799424/216951581-fcc7e30d-36cc-4aa6-a506-354153f6929f.png)

А при большом все равно пришлось бы менять дизайн.

Кстати, глядя на поганый вид текста после теста на переполнение, я бы не считал лишним для таких фоновых блоков на всякий случай прописывать белый цвет.
За примером далеко ходить не надо. Полегла база картинок, либо медленный инет, либо юзер отключил картинки из экономии трафика (в путешествии иногда так делается) - и уже альтернативный текст фиг прочитаешь:
![изображение](https://user-images.githubusercontent.com/26799424/216952308-990df359-7dad-4eea-bfaa-535471880a73.png)

TromiUss replied:
Правда не понятно, как задать белый текст для alt y svg welcome
Цвет я задал, но при скрытии svg альт не появился, хотя скорее всего я просто неправильно воспользовался dev tools

efiand replied:
Не надо задавать цвет альту, надо тому блоку, которому мы задали фон.
Альт отнаследует цвет.

Если скрыть картинку, то мы и альта не увидим. Альт появляется, если картинка не загрузилась. Чтобы его воспроизвести, надо "испортить" путь к картинке:

![изображение](https://user-images.githubusercontent.com/26799424/217086167-d50a0cd6-5bd1-453b-9413-d35abbfe224c.png)
(прямо тут, в инспекторе текущего состояния разметки в браузере, можно редактировать HTML и сразу видеть, к чему это приведет)

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+
+}
+
+.advantages {
```
А я ведь специально рекомендовал `.content` как универсальную секцию, чтобы не писать три раза одинаковое поведение, например, центровку текста. Вы лид-тексты и обычные тексты переименовали в `content-*` - а секции почему-то нет.
Ну либо добавили бы второй класс `content` к ним, чтобы не дублировать под разными именами одну и ту же логику

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+.content {
+  text-align: center;
+  color: #ffffff;
+  background: url(../images/footer-background.jpg) no-repeat top center;
```
`.content` судя по имени подходит как враппер для `.content-title` и т. п.
Логику фона стоит вешать не на него, а на доп. класс - тут либо собственный `subscribe` какой-нибудь, либо `.content-accent` как **вариант** контентного блока с фоном и белым текстом.

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+  background-size: 100% auto;
+}
+
+.content-teased {
```
👍🏻  ну да, вот тот самый вариант контентного блока с белым текстом

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+.footer-links {
+  display:flex;
+  justify-content: space-between;
+  vertical-align: middle;
```
Для выравнивания флекса используется `align-items`

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+}
+
+.social-network-item {
+  display: inline;
```
Это жуткий антипаттерн с учетом наличия в современном мире флексов. Инлайновая сетка порождает пробелы между элементами. Вместо того, чтобы с этим бороться, проще флексы использовать.

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+
+.social-network-item {
+  display: inline;
+  text-align: center;
```
Ни на что не повлияет в конечной реализации

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+  display: inline;
+  text-align: center;
+  text-decoration: none;
+  margin: 0%;
```
У `<li>` нет маргинов и так, если специально не задавать

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+.social-network-item {
+  display: inline;
+  text-align: center;
+  text-decoration: none;
```
Это имело бы смысл вешать на ссылку, если бы у нее был видимый текст в этом блоке

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+.tel {
+  font-size: 40px;
+  line-height: 40px;
+  text-align: center;
```
Бесполезно при space-between

![изображение](https://user-images.githubusercontent.com/26799424/216954923-01221aa3-55f9-4191-892f-67a0726b748c.png)

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+}
+
+.html-academy-link {
+  width: 114.9px;
```
Это не нужно, так как ссылка охватит картинку, у которой размеры уже заданы. Если вам нужно победить белую полоску под картинкой, сделайте ее блочной

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+
+.wk {
+  background-image: url(../images/wk.svg);
+  background-position: center;
```
Сейчас вы задублировали только два свойства. Позже появятся и другие, в том числе состояния наведения, фокуса, нажатия. Самое время уже сейчас вспомнить, что есть `.social-network-link` для описания общих свойств

efiand commented on styles/styles.css:
```diff
@@ -42,3 +42,173 @@ body {
+}
+
+.visually-hidden {
+  visibility: hidden;
```
Это скрытие недоступно для читалок. Используйте рекомендации Академии или 
```
position: fixed;
transform: scale(0);
```

efiand replied:
👍🏻 А то, что прописали visually-hidden в конце, хорошая практика. Класс универсальный, не должен ничем случайно перебиться

efiand commented on styles/styles.css:
```diff
@@ -59,23 +59,23 @@ body {
+  font-display: swap;
+}
 
+/*---------------------HEADER---------------------*/
```
`body` это же не хедер. Не смущай вселенную, сдвинь плашку комментария ниже )

efiand commented on styles/styles.css:
```diff
@@ -85,11 +85,35 @@ body {
+
+.header-item {
+  display: inline;
+  text-align: left;
```
Это и так дефолтное значение

efiand commented on styles/styles.css:
```diff
@@ -85,11 +85,35 @@ body {
+                                                                До конца не разобрался еще, лучше оставлю так
+
+.header-item {
+  display: inline;
```
Спойлер: забудь инлайновый контекст форматирования для всего, кроме собственно текста (всякие там смайлики в тексте или шильдики). Все списки ссылок, в том числе иконочных, хлебные крошки, хэштеги (если они по дизайну не через единичный пробел) - это всё микросетки флексовые

efiand commented on styles/styles.css:
```diff
@@ -85,11 +85,35 @@ body {
+
+.user-item {
+  display: inline;
+  text-align: right;
```
Вместо этого всему будущему флекс-контейнеру задай `justify-content: flex-end`

efiand commented on styles/styles.css:
```diff
@@ -85,11 +85,35 @@ body {
 .header-page {
-  background-color: blue;
+  display:flex;
+  justify-content: baseline;
```
Базовая линия актуальна для выравнивания по вертикали. Но тут - нет, помня о кликабельной пустоте вокруг ссылок, чтобы до самой нижней линии была зона ссылки.

efiand commented on styles/styles.css:
```diff
@@ -85,11 +85,35 @@ body {
-  background-color: blue;
+  display:flex;
+  justify-content: baseline;
+  vertical-align: middle;
```
Для флексов используется align-items.
Тут, возможно, `align-items: center` подошел бы. Но так как почти никто не любит, чтобы при добавлении второго ряда ссылок лого ехало вниз, то скорее всего `align-items: flex-start`

efiand commented on styles/styles.css:
```diff
@@ -85,11 +85,35 @@ body {
+.user-item {
+  display: inline;
+  text-align: right;
+  margin: 0%;
```
И маргинов у `li` нет ни в одном браузере

efiand commented on styles/styles.css:
```diff
@@ -104,10 +128,23 @@ body {
+}
+
+.user-link-button:hover {
+  font-size: 16px;
```
Не надо в псевдоклассах состояния писать те свойства, которые не меняются при переходе к этому состоянию. Тут остаётся только background-color.

efiand commented on styles/styles.css:
```diff
@@ -124,6 +161,18 @@ body {
 }
 
+.user-count {
+  font-family: "PT sans";
```
Шрифт наследуется. Не надо писать его везде подряд.

efiand commented on styles/styles.css:
```diff
@@ -124,6 +161,18 @@ body {
+  font-weight: 400;
+  font-size: 10px;
+  line-height: 10px;
+  background-color: #7DB54F;
```
Цвета принято писать в каком-то одном регистре. Например, в нижнем, как того требует кодгайд Академии.

efiand commented on styles/styles.css:
```diff
@@ -124,6 +161,18 @@ body {
 
+.user-count {
+  font-family: "PT sans";
+  font-style: normal;
```
А мы где-то задействовали курсив?

efiand commented on styles/styles.css:
```diff
@@ -124,6 +161,18 @@ body {
+  line-height: 10px;
+  background-color: #7DB54F;
+  border-radius: 10px;
+  background-size: 20px 20px;
```
Там сплошной зеленый цвет фона, никакого изображения, зачем ему размер?

efiand commented on styles/styles.css:
```diff
@@ -124,6 +161,18 @@ body {
+.user-count {
+  font-family: "PT sans";
+  font-style: normal;
+  font-weight: 400;
```
Раз жирность больше на задана всему хедеру, это тоже теперь мусор.

![изображение](https://user-images.githubusercontent.com/26799424/216959241-0d379b6a-8e70-493e-961e-cb7b1c25d26d.png)
Вот, ничего не поменялось после отключения. А вот цвет идеологически нужен именно этому классу, а не ссылке, ссылка ведь на белом фоне. Захочет заказчик убрать скрытость спана с текстом - и будет горе

efiand commented on styles/styles.css:
```diff
@@ -133,22 +182,129 @@ body {
   background-size: 100% auto;
   text-align: center;
-  background-attachment: fixed;
+  background-attachment: scroll;
```
Это дефолтное значение, то есть мусор. Знаете о нем - и хорошо. Когда понадобится, поймете. Мне за 5 лет ни разу не понадобилось

efiand commented on styles/styles.css:
```diff
@@ -133,22 +182,129 @@ body {
 
+.advantages .content-title {
+  font-family: "PT sans";
+  font-style: normal;
```
Подытожу: не надо в проекте, где всего один шрифт и одно начертание, повторять его на всех элементах. Как только появятся изменения дизайна, будете переопределять.

efiand commented on styles/styles.css:
```diff
@@ -133,22 +182,129 @@ body {
   text-align: center;
 }
 
+.advantages .content-title {
```
В том и идея общих классов, что они везде ведут себя одинаково, иначе зачем они общие? А значит, что не надо писать вложенный селектор, это затруднит дальнейшую поддержку. Просто `.content-title`

efiand commented on styles/styles.css:
```diff
@@ -133,22 +182,129 @@ body {
+  line-height: 26px;
+}
+
+.advantages-item .advantages-title {           /*Как лучше, тег с классом ..-title в теге с классом ..-item или в теге с классом ..-list*/
```
`-list` это про список. Если заголовок наследует что-то от списка, можно на классе списка задать стили.
Для самого заголовка тут ни к чему вложенность, ведь он у нас сам по себе такой. Почему, например, не написать `body main .advantages .advantages-item .advantages-title {}` - поняли, к чему я клоню?

efiand replied:
Задание для самых пытливых: узреть, а advantages-title и services-title часом не одно и то же? И тексты под ними - по той же логике. Сетку карточек опять же можно посмотреть у обоих списков.

efiand commented on styles/styles.css:
```diff
@@ -133,22 +182,129 @@ body {
+.advantages-item .advantages-description {    /*Аналогично*/
+  font-family: "PT sans";
+  font-style: normal;
+  font-weight: 400;
```
А оно и было 400

efiand commented on styles/styles.css:
```diff
@@ -133,22 +182,129 @@ body {
   text-align: center;
 }
 
+.services .content-title {
```
![изображение](https://user-images.githubusercontent.com/26799424/216961050-bfe7f8d4-21a7-4647-a59c-bb6926ddbef0.png)

Давайте напишем рядом, удалим мусор и добавим нужное-забытое. Одно и то же. Значит, достаточно оставить просто `.content-title` без всякой вложенности.

efiand commented on styles/styles.css:
```diff
@@ -133,22 +182,129 @@ body {
+}
+                                    /*В этих двух селекторах можно было задать одинаковые классы (у тегов)
+                                    с content-text и -title и задать всем в двух селекторах свойства ( без уточнения второго класса ), тк там все одинаково
+                                    но думаю, если, например, заказчик попросит поменять св-ва текста в одном из этих блоков, придется прописывать доп. классы
```
В правильном ключе думаете.
Но именно поэтому названия `question` и `proposal` могут перестать быть актуальными, если заказчик поменяет контент. Что бывает реже, чем кардинальная смена дизайна.

На вопрос "а что если" - универсальный ответ всегда: если бы бабушка имела маленькую штучку, она была бы дедушкой. Того принципа, что одинаковые вещи делаем одинаково, это не отменяет: как только дизайн какого-то блока перестает описываться каким-то обшим классом, тогда класс и меняем. Либо добавляем еще класс, если изменения несущественные. Это всегда можно сделать, когда потребуется при конкретных правках, зачем заранее множить сущности.
Тогда почему бы не подумать так вообще про любой элемент, что в нем могут попросить частные правки? И написать 100 км стилей. Нет, лучше менять класс ровно тогда, когда элемент перестал быть тем, чем он был.
И если на то пошло, то в моей практике как раз чаще всего правки стилей распространялись на все идентичные элементы.

efiand commented on styles/styles.css:
```diff
@@ -133,22 +182,129 @@ body {
+}
+
+.button-big {
+  color: #ffffff;
```
Все текстовые кнопки и так белые.

TromiUss replied:
При удалении свойства текст стал черным, что не соответствует макету в Фигме
Не совсем понял, как тогда лучше

efiand replied:
Перенести на основной класс `button` задание цвета. На нем надо будет задать еще много чего, так как используется и для ссылок, и для кнопок - `box-sizing: border-box` для уравнивания поведения кнопок и ссылок, `border: none` для сброса у кнопок, `text-align: center` для использования с разной шириной, чтобы не задавать разные паддинги, так как отступы сбоку при заданной ширине зависят от ширины текста (а при незаданной ширине отступы одинаковые, но ширина зависит от текста), `font-family: inherit` для явного наследования кнопками шрифта; сбросить подчеркивание у ссылок. Заказчик может попросить лапку курсора на кнопки, тогда явно задать `cursor: pointer`. И часто рекомендуют `user-select: none;` - чтобы при выделении текста в него кнопки не попадали, они ведь служебные.
И поэтому `.button-big`, чтобы в нем не дублировать все эти свойства, рекомендуется делать добавочным классом. Соответственно, цвет задан в `.button` - и для маленьких кнопок, и для больших.

efiand commented on styles/styles.css:
```diff
@@ -133,22 +182,129 @@ body {
+  text-transform: uppercase;
+}
+
+.button-big {
```
big это больше про размер шрифта и паддинги. Возможно, про особый бордер-радиус. Про цвета это primary/secondary, accent/alternative, на худой конец blue/brown

efiand commented on styles/styles.css:
```diff
@@ -133,22 +182,129 @@ body {
+.button-big:hover {
+  color: #ffffff;
+  background-color: #927c71;
+  border-radius: 4px;
```
Разве радиус скругления такой только при нведении? Нет, он и так на `.button` уже задан.

efiand commented on styles/styles.css:
```diff
@@ -133,22 +182,129 @@ body {
+}
+
+.button-big:hover {
+  color: #ffffff;
```
Цвет мы задали для кнопок. Если при наведении он не меняется, не надо его дублировать

efiand commented on styles/styles.css:
```diff
@@ -133,22 +182,129 @@ body {
+
 .content {
   text-align: center;
   color: #ffffff;
```
Это всё уникальное оформление (фон, цвет) подошло бы не для абы какого контента, а только для того, который `-teased`. Та же секция подписки на внутренней странице уже не `-teased`, там бы было достаточно просто не добавлять второй класс.

efiand commented on styles/styles.css:
```diff
@@ -313,15 +318,52 @@ body {
 }
 
-/*---------------------FOOTER---------------------*/
+.content .content-title {
```
Мы для того и назвали класс с нужным префиксом, чтобы использовать его без дополнительной вложенности

efiand commented on styles/styles.css:
```diff
@@ -313,7 +268,7 @@ body {
   text-align: center;
   color: #ffffff;
-  background: url(../images/footer-background.jpg) no-repeat top center;
+  background: url(../images/footer-background.jpg) no-repeat top center;    /*Не совсем понял, как растянуть изображение как в hero*/
```
Сейчас у вас мешанина из кратой и подробной формы. Выберите один путь.
И в зависимости от выбора у вас два варианта записи решения:

```css
background: url("../images/footer-background.jpg") cover #242424;


background-image: url("../images/footer-background.jpg");
background-size: cover;
background-color: #242424;
```

По моему опыту, отдельная запись свойств надежнее. И поддерживать проще потом.

efiand commented on styles/styles.css:
```diff
@@ -378,6 +333,7 @@ body {
   line-height: 40px;
   text-align: center;
   text-decoration: none;
+  color: #242424;
```
Вы нашли более мощный способ с inherit, используйте его везде, не прогадаете.

efiand reviewed (commented):
Очень хорошо поработали над ошибками. По стилизации в целом неплохо, но есть проблемы, которые в основном сводятся к двум:

1. Пишете ненужные свойства
2. Дублируете, не принимая во внимание наследование как инструмент.

Проблема дублирования из-за того, что одинаковые вещи делаете под разными именами, чуть менее критична, но тоже стоит овладеть комбинациями классов для распределения сходств и отличий.

TromiUss committed:
fix

Исправил недочеты, кроме выравнивания списков

TromiUss committed:
вопрос

не могу понять, как сдвинуть второй список (user) в правую часть шапки

TromiUss committed:
Стилизация-2

Немного справил ссылки в шапке и в подвале, но сломал кнопку в header...

Список с преимуществами пока не смог оформить, как и иконки с едой, жильем и сувенирами

TromiUss committed:
Advantages

Получилось разделить эту секцию, но не совсем понял один момент.

При добавлении общего класса advantages-item элементам с картинкой все блоки флекс контейнера начинают игнорировать фотографии и залезать на них

TromiUss committed:
Сетка крупных блоков

TromiUss committed:
картинки в сервисах

TromiUss committed:
Исправления advantages

TromiUss committed:
отступы

TromiUss committed:
фоновый цвет

efiand commented on index.html:
```diff
@@ -56,25 +56,28 @@ <h2 class="visually-hidden">Преимущества.</h2>
           <li class="advantages-item advantages-item-wide">
-            <h3 class="advantages-title">Настоящий городок</h3>
-            <p class="advantages-description">Седона — не аттракцион для туристов, там течёт своя жизнь</p>
+            <h3 class="advantages-title block-title ">Настоящий городок</h3>
```
Единообразие оформления требует не оставлять пустых пробелов перед кавычками. и чтобы голова о том не думала, надо использовать автоформатирование `alt + shift + f`. Если у вас это не работает, дайте знать, починим, но так нельзя оставлять

efiand commented on index.html:
```diff
@@ -56,25 +56,28 @@ <h2 class="visually-hidden">Преимущества.</h2>
           <li class="advantages-item">
-            <h3 class="advantages-title">Небольшая площадь</h3>
-            <p class="advantages-description">Все достопримечательности находятся очень близко</p>
+            <h3 class="advantages-title block-title">Небольшая площадь</h3>
```
`advantages-title` и `advantages-description` не принимают участия в стилизации - нужны ли они?

efiand commented on index.html:
```diff
@@ -85,30 +88,37 @@ <h2 class="visually-hidden">Сервисы.</h2>
           <li class="services-item">
-            <h3 class="services-title">Жильё</h3>
-            <p class="services-description">Рекомендуем пожить в настоящем мотеле, всё как в кино!</p>
+            <h3 class="services-title block-title">Жильё</h3>
```
`services-title` и `services-description` не принимают участия в стилизации - нужны ли они?

efiand commented on index.html:
```diff
@@ -85,30 +88,37 @@ <h2 class="visually-hidden">Сервисы.</h2>
-        <b class="hotels-question">Заинтересовались?</b>
-        <p class="hotels-proposal">Укажите предполагаемые даты поездки, и мы покажем вам лучшие предложения гостиниц в
-          Седоне</p>
+        <b class="content-title question">Заинтересовались?</b>
```
Нужен ли класс `question` (да еще такой опасный в плане слишком универсального названия) - если учесть, что **ВСЕ** `content-title` и так по макету uppercase? По макету нужно было сделать все шрифты и цвета, не сделано.

efiand commented on index.html:
```diff
@@ -85,30 +88,37 @@ <h2 class="visually-hidden">Сервисы.</h2>
     <footer class="footer-page">
       <section class="content content-teased">
-        <h2 class="content-title">Подпишитесь на рассылку</h2>
+        <h2 class="content-title subscribe">Подпишитесь на рассылку</h2>
```
Нужен ли класс `subscribe` - если учесть, что **ВСЕ** `content-title` и так по макету uppercase?

efiand commented on index.html:
```diff
@@ -121,18 +131,24 @@ <h2 class="content-title">Подпишитесь на рассылку</h2>
         <a class="tel" href="tel:+78128121212">+7 (812) 812-12-12</a>
         <a href="https://htmlacademy.ru" class="html-academy-link">
-          <img src="images/htmlacademy-logo.svg" width="114.9" height="32.9" alt="Логотип HTML Academy">
+          <img src="images/htmlacademy-logo.svg" alt="Логотип HTML Academy">
```
То, что внутри иконки размеры правильные целочесленные, не отменяет того, что для уменьшение скачков контента на странице мы задаем картинкам размеры в самом коде этой страницы. Критерий же не перестал требовать размеров, вы ведь их читаете регулярно, чтобы по ним научиться работать? В общем, надо скопировать из svg-файла размеры и вставить сюда.

efiand replied:
alt по-прежнему нуждается в точке на конце

efiand commented on styles/styles.css:
```diff
@@ -1,364 +1,200 @@
-  font-style: italic;
   font-weight: 400;
-  src: url(../fonts/ptsans-400.woff2) format(woff2);
+  src: url("../fonts/ptsans-400.woff2") format(woff2);
```
Если делать кавычки (что хорошо) - то делать везде. Ниже для картинок это не сделано

efiand commented on styles/styles.css:
```diff
@@ -1,364 +1,200 @@
+  font-family: "PT Sans", sans-serif;
   font-size: 16px;
-  line-height: 24px;
+  line-height: 1.5%;
```
В процентах это 150%. Но проценты могут создавать проблемы https://developer.mozilla.org/ru/docs/Web/CSS/line-height - пишите, как я и рекомендовал, множителем, то есть просто `1.5` без единиц измерения

efiand commented on styles/styles.css:
```diff
@@ -1,364 +1,200 @@
-
-.advantages {
+  background-image: url(../images/index-background.jpg);
+  background-color: inherit;
```
![изображение](https://user-images.githubusercontent.com/26799424/218438242-bad3bbef-5810-400c-99ce-39ce498af1ca.png)

Зачем мусорить? Чтобы такого не происходило, следует соблюдать порядок свойств. Есть рекомендации академии, сам пользуюсь: https://codeguide.academy/html-css.html#css-order

efiand commented on styles/styles.css:
```diff
@@ -1,364 +1,200 @@
+  background-image: url(../images/index-background.jpg);
+  background-color: inherit;
+  background-position: center;
+  background-repeat: no-repeat;
```
При `cover` указание неповторяемости теряет смысл. Позиционирование - во многих случаях тоже (его следует задавать при `cover` тогда, когда фон особый или не решена задача попадания в макет (это будет видно уже после этапа сеток)

efiand replied:
При `cover` указание неповторяемости теряет смысл. Позиционирование - во многих случаях тоже (его следует задавать при `cover` тогда, когда фон особый или не решена задача попадания в макет (это будет видно уже после этапа сеток)

efiand commented on styles/styles.css:
```diff
@@ -1,364 +1,200 @@
 /*---------------------FOOTER---------------------*/
 
-.content {
+.content-teased {
```
`.content` (как родитель `.content-title`, `.content-description` мы видим в четырех местах. И шрифтовые свойства у него одинаковые. `-teased` это только про цвет и фон, ведь будет скоро и рассылка на внутренней странице.)
Рассылка остаётся типовой контентной секцией в отношении поведения потомков и в отношении отступов, и поэтому у нее есть класс `.content`. Но в то же время она имеет особую тему оформления на главной, и поэтому ей можно дать на главной **добавочный** класс `content-teased`.

efiand commented on styles/styles.css:
```diff
@@ -1,364 +1,200 @@
   background-color: #82B3D3;
   border-radius: 0px 4px 4px 0px;
-  color: #ffffff;
+  color: inherit;
```
Тут кнопка белая не потому, что она в белом блоке рассылки (на внутренней-то блок не белый) - а потому, что фон голубой, поэтому было правильно задавать явно белый цвет.

efiand commented on styles/styles.css:
```diff
@@ -1,364 +1,200 @@
-}
-
 .submit {
-  font-family: "PT sans";
```
Шрифт кнопкам `<button>` задавать надо, но чтобы не писать каждый раз `"PT Sans", sans-serif`, можно `inherit`.

efiand commented on styles/styles.css:
```diff
@@ -1,364 +1,200 @@
   line-height: 36px;
   text-transform: uppercase;
   background-color: #82B3D3;
   border-radius: 0px 4px 4px 0px;
```
У нулевого значения не указывают единицы измерения

efiand commented on styles/styles.css:
```diff
@@ -1,364 +1,200 @@
   display: flex;
   justify-content: space-between;
-  vertical-align: middle;
+  align-items: middle;
```
`align-items: center;`

efiand commented on styles/styles.css:
```diff
@@ -1,364 +1,200 @@
 
-.YouTube {
+.youtube {
   background-image: url(../images/YT.svg);
```
Не решена проблема с переименованием иконки: она большими буквами так и будет. Назовите `youtube.svg` и не парьтесь. Я бы понял, если бы соседняя была именно `tg.svg`, и захотелось бы побороться за двухбуквенность. Но нет же.

efiand commented on styles/styles.css:
```diff
@@ -24,10 +24,33 @@ body {
 }
```
А зачем на `body` цвет 242424? Большинство блоков обычного цвета по умолчанию.

efiand commented on styles/styles.css:
```diff
@@ -24,10 +24,33 @@ body {
+  display: flex;
+  justify-content: right;
+  margin-left: auto;
+  width: 268px;
```
Маленькая очень ширина, задавайте с запасом, ничего не влезает
![изображение](https://user-images.githubusercontent.com/26799424/218441599-f71a0ad9-2fc3-4335-9448-d8817af00143.png)

Конечно, Академия всех озадачила, что вы сначала на флексах напилите костылей, а потом осознаете, что в большинстве случаев распределение ширины было удобнее на гридах :-(

efiand commented on index.html:
```diff
@@ -84,23 +82,28 @@ <h3 class="advantages-title block-title">Там есть мост дьявола
-        <h2 class="visually-hidden">Сервисы.</h2>
-        <b class="content-title">Приезжайте в Седону отдохнуть в комфорте и уюте!</b>
-        <p class="content-text">Опытный персонал и качественное обслуживание!</p>
+        <div class="advices">
```
Лучше без лишних оберток. Если для проверяющего это боль достаточной силы, он может отклонить по критерию. Тут всё рулится верхним паддингом этой секции или нижним маргином предыдущей.

Если и заводить такой враппер, то:
* Задавать все 4 паддинга, так как от касания текста в боковые в края сейчас спасает только то, что текста не так много.
* Повесить универсально на все 4 подобные секции `content` в виде `<header class="content-header">` - тогда и семантика есть (вроде как не лишняя обертка-то) - и универсальность поведения (text-align: center не придется 4 раза писать в css). А то, что между некоторыми секциями больше места, то можно будет догнать нижними маргинами секций. У нас минимальный паддинг 54px:
![изображение](https://user-images.githubusercontent.com/26799424/218443228-9932615d-db02-45e7-8cd9-dd5cc400ca70.png)

efiand commented on index.html:
```diff
@@ -84,23 +82,28 @@ <h3 class="advantages-title block-title">Там есть мост дьявола
+        <div class="advices">
+          <h2 class="visually-hidden">Сервисы.</h2>
+          <b class="content-title">Приезжайте в Седону отдохнуть в комфорте и уюте!</b>
+          <p class="content-text damn">Опытный персонал и качественное обслуживание!</p>
```
`damn` не нужен, так как тут правильнее было бы задавать нижний маргин `content-title`. И обнулять маргины у `content-text`, это же одинаковое поведение во всех четырех таких блоках.

Почему-то поначалу понятно, что если одинаковые блоки рядом, то их надо одинаково стилизовать, а если разбросаны, то почему-то выдумывается 4 разных решения одной и той же задачи:
![изображение](https://user-images.githubusercontent.com/26799424/218444317-6202a02e-b98a-489d-a640-e16081f5b2bb.png)

efiand commented on index.html:
```diff
@@ -84,23 +82,28 @@ <h3 class="advantages-title block-title">Там есть мост дьявола
+        </div>
         <ul class="services-list">
           <li class="services-item">
+            <span class="visually-hidden house">Жилье.</span>
```
Зачем, во-первых, дублировать текст, который и так есть в заголовке, а, во-вторых, еще и давать второй класс блоку, задачи которого решаются (тут решаются зачем-то) классом `visually-hidden`.
Если это попытки стилизовать фоны декоративной графики, то надо было слушать авторов и читать внимательно программу, не случайно же пока говорится, что не надо трогать до поры декоративную векторную графику. Потому что псевдоэлементы будут вам давать, которые и нужны для этой задачи, а сейчас вы изобретаете ненужный велосипед

efiand commented on index.html:
```diff
@@ -122,7 +125,7 @@ <h2 class="visually-hidden">Поиск отелей.</h2>
         <p class="content-text">Только полезная информация и никакого спама, честное бойскаутское!</p>
         <form method="post" action="https://echo.htmlacademy.ru">
-          <label>
+          <label class="mail-line">
```
Какие задачи решает этот класс?

efiand commented on styles/styles.css:
```diff
@@ -21,36 +20,32 @@ body {
 .header-navigation {
   display: flex;
+  align-items: center;
+  margin-left: auto;
```
Бесполезное свойство, учитывая наличие явного маргина у лого

efiand commented on styles/styles.css:
```diff
@@ -21,36 +20,32 @@ body {
-  margin-left: auto;
-  width: 268px;
-
+  justify-content: flex-start;
```
Дефолтное значение такое и есть, нет смысла задавать явно.

efiand commented on styles/styles.css:
```diff
@@ -59,9 +54,18 @@ body {
   line-height: 24px;
   color: inherit;
   text-decoration: none;
+  padding: 20px 16px;
```
Сейчас ссылки "вывыалились" из li по вертикали. Не очень хорошее поведение, таит потенциальную опасность. Чтобы такого не было, сделайте их блочными.

efiand commented on styles/styles.css:
```diff
@@ -59,9 +54,18 @@ body {
+  display: flex;
+  justify-content: flex-end;
+  width: 268px;
+  margin: 0;
```
Чтобы оттолкнулся вправо, нужен левый автомаргин. Иначе придется вымерять соотношение ширин обоим спискам, чтобы кнопка не резалась

efiand commented on styles/styles.css:
```diff
@@ -127,11 +138,39 @@ body {
 }
 
+.content-main {
+width: 620px;
```
Ни к чему. Ведь там центровка всё решает. А вот паддинги для защиты от касания в края стоит предусмотреть. Либо через обертку (см. про совет насчет `content-header`) - либо на данном блоке

efiand commented on styles/styles.css:
```diff
@@ -127,11 +138,39 @@ body {
 }
 
+.services-list {
+  display: flex;
```
Карточек может быть больше, нужно flex-wrap.

efiand commented on styles/styles.css:
```diff
@@ -127,11 +138,39 @@ body {
+.services-list {
+  display: flex;
+  list-style: none;
+  vertical-align: middle;
```
![изображение](https://user-images.githubusercontent.com/26799424/218446571-d0445fdc-f5e2-4d8a-bdb9-3c0edda26aa4.png)
Тут при росте одной карточки в высоту должны будут вырастать и остальные:
![изображение](https://user-images.githubusercontent.com/26799424/218446917-306d5386-74c9-411c-89f1-2af1937fe415.png)


поэтому не надо вертикальное выравнивание вообще трогать

efiand commented on styles/styles.css:
```diff
@@ -127,11 +138,39 @@ body {
+
+.services-item {
+  width: 230px;
+  height: 223px;
```
https://up.htmlacademy.ru/htmlcss/36/criteries#TEST-04
Лучше спросите, в чем смысл данного критерия, если нет понимания. Веб живой, текст никогда не будет всегда как по макету, и отныне (когда вы открыли для себя этап сеток) и всегда, чтобы не переделывать проекты много раз, помните и думайте о том, что текста может быть больше или меньше, чем по макету.
Поэтому высоту для текстовых блоков вообще никогда не задавайте! В крайнем случае - минимальную можно задать.

efiand commented on styles/styles.css:
```diff
@@ -127,11 +138,39 @@ body {
+  width: 230px;
+  height: 223px;
+  padding: 81px 85px;
+  background-position: center;
```
center по горизонтали, top в пикселях по вертикали - чтобы надеть фоновые иконки на то место, где нужно предусмотреть увеличенный верхний паддинг (это если псевдоэлементы не хочется применять)

efiand commented on styles/styles.css:
```diff
@@ -143,13 +182,30 @@ body {
+}
+
+.hotels-search {
+  padding: 96px 304px;
```
Для кнопок лучше задать одинаковый паддинг (горизонтальный по самой узкой кнопке) и центровку по текста по горизонтали - а дальше регулировать через ширину. Ведь кнопка коричневая тут и синяя в форме больше ширины и цвета ничем не отличается.

Что мы видим:
* Белотекстовые жирные кнопки апперкейс с идентичным поведением на ховере и нажатии - и при том с дофига стилей по сбросу браузерных умолчаний
* Два варианта цвета - коричневый и синий.
* Два варианта размера шрифта и паддингов.

Это все можно обыгрывать 1-3 классами, например:
1. `button` на всех кнопках с белым жирным апперкейсным текстом
2. `button-primary` для более важной кнопки, то есть переопределение голубого на коричневый и `button-reset` для сброса фона кнопки ресета в фильтрах каталога
3. `button-big` для увеличенного шрифта и паддингов.
4. Личные классы для уникального поведения в родителе (имена от имен классов родителя, например, ширина на `search-button`, бордер-радиусы на `subscribe-button`

2 и 3 называются модификаторами, 4 - миксами, и речь не только о БЭМ

efiand commented on styles/styles.css:
```diff
@@ -185,13 +243,17 @@ body {
-  margin: 60px;
+  align-items: center;
+  padding: 40px 70px 30px 70px;
+
```
Не оставляйте ненужных пустых строк. Разумеется, об этом не надо думать постоянно, нужно просто нажимать `alt + shift + f`

efiand commented on styles/styles.css:
```diff
@@ -185,13 +243,17 @@ body {
-  align-items: middle;
-  margin: 60px;
+  align-items: center;
+  padding: 40px 70px 30px 70px;
```
Четвертое значение, равное второму, не принято указывать

efiand commented on styles/styles.css:
```diff
@@ -185,13 +243,17 @@ body {
   display: flex;
-  text-decoration: none;
+  justify-content: flex-start ;
+  width: 142px;
```
Может понадобиться добавить соц. ссылок, я бы с запасом оставил: расстояния до ссылки на телефон позволяют. Учитывайте, что для центровки телефонной ссылки нужно сделать одинакового размера первый и третий блоки

efiand commented on styles/styles.css:
```diff
@@ -185,13 +243,17 @@ body {
+  justify-content: flex-start ;
+  width: 142px;
+  list-style: none;
+  padding: 0;
```
Маргины тоже не забывайте сбрасывать, от браузерных отступов лучше не зависеть.

efiand commented on styles/styles.css:
```diff
@@ -219,6 +281,5 @@ body {
 .visually-hidden {
-  position: fixed;
-  transform: scale(0);
+  visibility: hidden;
```
Вот зачем опять вернули? Пока не поймете роль класса `.visually-hidden`, дичь твориться и будет

efiand commented on index.html:
```diff
@@ -54,11 +54,14 @@ <h2 class="visually-hidden">Преимущества.</h2>
-            <p class="advantages-description block-description">Седона — не аттракцион для туристов, там течёт своя
-              жизнь</p>
-            <img src="images/photo-1.jpg" width="800" height="384" alt="Город Седона.">
+          <li class="advantages-item-wide">
```
Классы, у которых начальные части названия похожи, намекают на сходство. Но на самом деле ведь это разные вещи. `advantages-group`, например.

> При добавлении общего класса advantages-item элементам с картинкой

Так не надо пытаться разные вещи реализовывать одинаково. Классы - это не про семантику, это про стилизацию. У стилистически разных элементов не должно быть одинаковых классов. Верно и обратное, кстати.

efiand commented on index.html:
```diff
@@ -54,11 +54,14 @@ <h2 class="visually-hidden">Преимущества.</h2>
-              жизнь</p>
-            <img src="images/photo-1.jpg" width="800" height="384" alt="Город Седона.">
+          <li class="advantages-item-wide">
+            <section class="wide-photo">
```
Не советую использовать секцию в секции, это усложняет структуру. Здесь можно использовать обычный див, а для `content-block` - `<header>`, это же заголовочная часть секции по уму

efiand replied:
Класс плохой:
1. Фото где угодно, но не в этом блоке. Называйте вещи своими именами, иначе - усложнение поддержки проекта.
2. Ничего нового не внесли: **с точки зрения стилей** это не что иное как `advantages-item` - такая же карточка, только синенькая (при стилизации плевать на семантику и структуру, значит, и при нейминге классов - тоже). Можно застилизовать "скин оформления" через `.advantages-group .advantages-item {}` или развесить дополнительные классы для цветов.

efiand commented on index.html:
```diff
@@ -88,7 +93,7 @@ <h2 class="visually-hidden">Сервисы.</h2>
         </div>
         <ul class="services-list">
-          <li class="services-item">
+          <li class="services-item services-item-color">
```
Помним о переполнении: при добавлении карточек заказчик хочет красивой шахматной последовательности:
![изображение](https://user-images.githubusercontent.com/26799424/218451980-54b44e02-c45a-471c-adea-38cf4d93f505.png)

Поэтому цвета в обоих списках с карточками лучше задать через nth-child

efiand commented on index.html:
```diff
@@ -136,17 +141,17 @@ <h2 class="content-title subscribe">Подпишитесь на рассылку
           <li class="social-network-item">
             <a class="social-network-link vk" href="https://vk.com/htmlacademy">
-              <span class="visually-hidden">Вконтакте.</span>
+              <span class="visually-hidden">вкон.</span>
```
Это ради прикола, что ли? Было ж лучше

TromiUss replied:
Из-за длины слова изменялась ширина картинки, единственное, что нашел - сократить обозначение ссылок до одинакового кол-ва символов...

efiand replied:
Пока ты не поймешь, зачем ты делаешь что-либо, будут такие ситуации. Доступный текст мы оставляем, чтобы интерфейс был удобен для тех, кто не смотрит сайт, а слушает. Если более обще: содержимое страницы не должно никак затрагиваться реализацией стилей. Надо было подумать о том, что ты что-то не так делаешь, а именно - поставил опять странные стили для visually-hidden. Потому что не понял, что такое visually-hidden и какая у него цель. Можно на консультации коснуться этого вопроса.

efiand commented on styles/styles.css:
```diff
@@ -121,7 +122,7 @@ body {
   background-position: center;
   background-repeat: no-repeat;
-  background-size: cover;
+  background-size:cover;
```
Или автоформатируйте, или следите за руками пристальнее. Опять форматирование неединообразное.

efiand commented on styles/styles.css:
```diff
@@ -142,6 +143,29 @@ body {
 }
 
+.advantages {
+  height: 1154px;
```
Забудем о высоте для текстовых блоков. Там и так всё разместится потом за счет паддингов

efiand commented on styles/styles.css:
```diff
@@ -142,6 +143,29 @@ body {
+.advantages-item-wide {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
```
Не соглашусь. При добавлении текста картинка вряд ли должна висеть по центру
![изображение](https://user-images.githubusercontent.com/26799424/218452768-fe330367-f702-4fcb-9a4d-d6c3b2612bf3.png)

По крайней мере при попытке такое сдать в Лигу А ни один тестировщик не примет. А лучше так:

![изображение](https://user-images.githubusercontent.com/26799424/218452871-5badcde0-28cd-41a0-b3c8-f9b2f360fc56.png)
(для картинки `height: 100%; object-fit: cover;`)

efiand commented on styles/styles.css:
```diff
@@ -142,6 +143,29 @@ body {
+
+.advantages-item {
+  width: 400px;
+  height: 385px;
```
Возможно, min-height тут понадобится, но height плохо: текст может выпадать:

![изображение](https://user-images.githubusercontent.com/26799424/218453899-3915b06d-e32a-422c-b857-d39ed6a6b44b.png)

efiand commented on styles/styles.css:
```diff
@@ -162,6 +188,10 @@ width: 620px;
   background-repeat: no-repeat;
 }
 
+.services-item-color {
```
Они все в той или иной степени `color` ) Некоторые светлее или темнее, от этого можно оттолкнуться в названии. Но рекомендую `.services-item:nth-child(even) {}` для задания иного фона

efiand commented on styles/styles.css:
```diff
@@ -250,7 +280,7 @@ width: 620px;
 .social-network {
   display: flex;
-  justify-content: flex-start ;
+  justify-content:space-between;
```
Ссылки плотно прилегают (если правильно сохранить иконки в фигме
![изображение](https://user-images.githubusercontent.com/26799424/218454353-4da8823a-6470-4055-a8ce-731d05bd27ea.png)

и ожидается возможность добавления справа еще нескольких. Поэтому спейс тут не подходит

efiand commented on index.html:
```diff
@@ -127,14 +127,16 @@ <h2 class="visually-hidden">Поиск отелей.</h2>
+          <h2 class="content-title subscribe">Подпишитесь на рассылку</h2>
+          <p class="content-text">Только полезная информация и никакого спама, честное бойскаутское!</p>
+        </div>
+        <form class="submit" method="post" action="https://echo.htmlacademy.ru">
```
Сабмит это про кнопку отправки. Назвать всю форму сабмитом - против здравого смысла.

efiand commented on styles/styles.css:
```diff
@@ -246,27 +247,47 @@ width: 620px;
+
+.subscribe-text {
+  margin-bottom: 54px;
+  padding-bottom: 54px;
```
Маргина тут достаточно

efiand commented on styles/styles.css:
```diff
@@ -246,27 +247,47 @@ width: 620px;
+
+.input {
+  border-style: none;
+  border-radius: 4px 0px 0px 4px;
```
Такой бордер-радиус не может быть у универсального инпута проекта (на что намекает название класса). Он тут исключительно по причине соседства с кнопкой. Как вариант, назвать `subscribe-input`. Такие частные стили на универсальных классах могут создавать баги, я сам видел пару раз

efiand commented on index.html:
```diff
@@ -95,20 +95,23 @@ <h2 class="visually-hidden">Сервисы.</h2>
         <ul class="services-list">
           <li class="services-item services-item-color">
             <span class="visually-hidden house">Жилье.</span>
+            <img class="services-images" src="images/living.svg"  alt="">
```
Если у вас что-то не получилось, странно думать, что мир изменился и теперь эти картинки внезапно "оконтентились". Нет, они по-прежнему фоновые :-)

(но если и так, зачем тогда выше мусорные спаны невидимые)

efiand commented on index.html:
```diff
@@ -75,7 +75,7 @@ <h3 class="advantages-title block-title">Красивая дорога</h3>
             <p class="advantages-description block-description">Большинство едет в Гранд Каньон и толпится там</p>
           </li>
-          <li class=" advantages-item-wide">
+          <li class=" advantages-item-wide item-wide-reverse">
```
Обычно дополнительные классы именуют созвучно с основными, например, `advantages-item-wide advantages-item-wide-reverse`. Но так как выше я писал, что тут имеем ложное созвучие с классами "карточек", советую `advantages-group advantages-group-reverse`

efiand commented on styles/styles.css:
```diff
@@ -154,18 +155,31 @@ width: 620px;
   padding: 0;
   margin: 0;
   list-style: none;
+  width:inherit;
```
Ни к чему. Там и так на всю ширину.

efiand commented on styles/styles.css:
```diff
@@ -146,6 +146,7 @@ width: 620px;
 .advantages {
   height: 1154px;
   margin-bottom:  126px;
+  width: inherit;
```
Ни к чему. Там и так на всю ширину.

efiand commented on styles/styles.css:
```diff
@@ -154,18 +155,31 @@ width: 620px;
-  justify-content: space-between;
   align-items: center;
+  margin: 0;
+  width: 1200px;
```
Лучше (с учетом того, что почти везде требуется адаптив) привыкать в таких случаях писать `100%`

efiand commented on styles/styles.css:
```diff
@@ -154,18 +155,31 @@ width: 620px;
+}
+
+.wide-photo {
+  margin: 101.2px 85px;
```
Лучше всегда краевые отступы родителя делать паддингами, наподобие полей у страницы в ворде.

efiand commented on styles/styles.css:
```diff
@@ -154,18 +155,31 @@ width: 620px;
   width: 400px;
   height: 385px;
-  align-content: center;
+  display: flex;
```
Тут вроде бы и просится вертикальная центровка, учитывая, что у всех типов карточек разные вертикальные паддинги, но при переполнении текстом оно некрасиво поедет вниз в соседних карточках, поэтому лучше не центровать тут вертикально. А значит, и флекс не нужен

efiand commented on styles/styles.css:
```diff
@@ -209,11 +223,15 @@ width: 620px;
   font-size: 24px;
   line-height: 28px;
+  width: 230px;
+  height: 28px;
```
Никогда не задавайте высоту контенту. Тут и так всё решает line-height для попадания однострочного текста в макет

efiand reviewed (commented):
Не везде до сих пор цвет приведен к нижнему регистру

TromiUss committed:
Исправления

TromiUss committed:
исправления 2

TromiUss committed:
Консультация

TromiUss committed:
Sticky-footer

TromiUss committed:
Update styles.css

TromiUss committed:
second-page

TromiUss committed:
2 page hero

TromiUss committed:
beta

TromiUss committed:
hero-2

TromiUss committed:
hotels-plate
